### ISSUE TITLE
Error - Fix Needed - System.Reflection.TargetInvocationException

### ISSUE THREAD

**Issue by DarickBrokaw** [DarickBrokaw/DTO_ML_RevitAddin#1]
I had to end my daily work. I committed and synced. Testing discovered this error on Revit Startup. ![System Reflection TargetInvocationException](https://user-images.githubusercontent.com/93608701/233109690-dfb1d96c-a42b-4ad7-8328-9678d426f632.png)

- May try to learn more about where the error is via catch logging 

```
public async void OnShutdown(ControlledApplication ctlApp)
{
    try
    {
//some code
    }
   catch (System.Reflection.TargetInvocationException ex)
    {
        if (ex.InnerException != null)
        {
            MessageBox.Show(ex.InnerException.Message);
        }
        else
        {
            MessageBox.Show(ex.Message);
        }
    }
    catch (Exception ex)
    {
        MessageBox.Show(ex.ToString());
    }
}
```

**Comment by DarickBrokaw** [comment]
I started debug mode and found this. The thing I do not understand is that
line 47:           `String LogFilePath = ConfigurationManager.OpenExeConfiguration(System.Reflection.Assembly.GetExecutingAssembly().Location).AppSettings.Settings["LogFilePath"].Value; // Log file path`
Works in the Main branch. Not sure what to do here...?
![image](https://user-images.githubusercontent.com/93608701/233633823-2cf292d9-0e30-4a86-9ef4-542c2b6b1a19.png)

**Comment by DarickBrokaw** [comment]
The reason for the above-posted error and its being null is that as seen in the commits, I added the reference for Nuget packages of System.Text.Json and Microsoft.Bcl.AsynInterfaces and this added to the DTO.dll.config file

```
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <runtime>
    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
      <dependentAssembly>
        <assemblyIdentity name="System.Runtime.CompilerServices.Unsafe" publicKeyToken="b03f5f7f11d50a3a" culture="neutral" />
        <bindingRedirect oldVersion="0.0.0.0-6.0.0.0" newVersion="6.0.0.0" />
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="Microsoft.Bcl.AsyncInterfaces" publicKeyToken="cc7b13ffcd2ddd51" culture="neutral" />
        <bindingRedirect oldVersion="0.0.0.0-8.0.0.0" newVersion="8.0.0.0" />
      </dependentAssembly>
    </assemblyBinding>
  </runtime>
</configuration>
```

Over writing 

```
<?xml version="1.0" encoding="utf-8" ?>
<configuration>
	<appSettings>
		<add key="LogFilePath" value="T:\WIP\`Logs\SPOT_Revit.csv" />
	</appSettings>
</configuration>
```

Need to find out how to fix this issue and possibly combine the two files into one .config file.

**Comment by DarickBrokaw** [comment]
You can't use System.Text.Json directly in a .Net 4.8 lib. System.Text.Json is .Net Core and is not compatible with .Net 4.8. Not sure if this will work with .Net Standard. Might be easier to just use Newtonsoft which ios very similar and supports .Net Framework .  [](https://stackoverflow.com/questions/68401655/system-text-json-deserialize-error-system-memoryextensions-cannot-load-system-ru#:~:text=You%20can%27t%20use%20System.Text.Json%20directly%20in%20a.Net%204.8,ios%20very%20similar%20and%20supports.Net%20Framework%20%E2%80%93%20Liam)

Upon closer inspection, https://www.nuget.org/packages/System.Text.Json/8.0.0-preview.3.23174.8#supportedframeworks-body-tab Does not highlight in the .NET Framework row net48 (assuming this is .NET Framework 4.8)

Still looking to understand this.

**Comment by DarickBrokaw** [comment]
See Fix in commit "cd9e98a946b5e2f0f6b90acdbff3d10f5a16d949"
